packages/design-system/src/components/Pagination/Pagination.test.tsx

```diff
@@ -0,0 +1,253 @@
+import './matchMedia.mock';
+import { mount, shallow } from 'enzyme';
+import Pagination from './Pagination';
+import React from 'react';
+
+describe('Pagination', () => {
+  const pageButtonSelector = '.ds-c-button';
+  const onPageChange = jest.fn();
+
+  const render = (overrideProps = {}, shouldDeepRender = false) => {
+    const props = {
+      totalPages: 3,
+      onPageChange: onPageChange,
+      renderHref: () => '/test',
+      ...overrideProps,
+    };
+
+    return shouldDeepRender ? mount(<Pagination {...props} />) : shallow(<Pagination {...props} />);
+  };
+
+  it('should render component', () => {
+    const wrapper = render({ currentPage: 2 });
+    expect(wrapper.is('nav')).toBe(true);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  describe('accessibility attributes', () => {
+    it('should have navigation label', () => {
+      const wrapper = render({ totalPages: 8 });
+      expect(wrapper.prop('aria-label')).toEqual('Pagination');
+    });
+    it('should set a custom navigation label', () => {
+      const wrapper = render({ totalPages: 8, ariaLabel: 'Pagey page page' });
+      expect(wrapper.prop('aria-label')).toEqual('Pagey page page');
+    });
+  });
+
+  it('should add custom className if specified', () => {
+    const customClassName = 'custom-class';
+    const wrapper = render({ totalPages: 8, className: customClassName });
+    expect(wrapper.hasClass(customClassName)).toBeTruthy();
+  });
+
+  describe('interactivity', () => {
+    describe('onPageChange', () => {
+      afterEach(() => {
+        jest.resetAllMocks();
+      });
+
+      it('should call onPageChange when "previous" is pressed', () => {
+        const wrapper = render({ currentPage: 2 });
+        wrapper.childAt(0).simulate('click', {});
+
+        expect(onPageChange).toHaveBeenCalledTimes(1);
+        expect(onPageChange).toHaveBeenCalledWith(expect.anything(), 1);
+      });
+
+      it('should call onPageChange when "next" is pressed', () => {
+        const wrapper = render({ currentPage: 2 });
+        wrapper.childAt(2).simulate('click', {});
+
+        expect(onPageChange).toHaveBeenCalledTimes(1);
+        expect(onPageChange).toHaveBeenCalledWith(expect.anything(), 3);
+      });
+
+      it('should call onPageChange when a page is pressed', () => {
+        const wrapper = render({ currentPage: 2 }, true);
+        wrapper.find('ul').childAt(0).find(pageButtonSelector).simulate('click', {});
+
+        expect(onPageChange).toHaveBeenCalledTimes(1);
+        expect(onPageChange).toHaveBeenCalledWith(expect.anything(), 1);
+      });
+    });
+
+    describe('href', () => {
```

Comment:
I like these two unit tests. Should there be one more for the href on one of the page buttons in the middle?